src/firefly/java/edu/caltech/ipac/astro/ibe/datasource/WiseIbeDataSource.java

```diff
@@ -128,11 +114,22 @@ public void initialize(Map<String, String> dsInfo) {
 
         String productLevel = dsInfo.get("ProductLevel");
 
+        List<String> imageSets = Arrays.asList(imageset.split(","));
+
         if (StringUtils.isEmpty(productLevel)) {
             String sourceId = dsInfo.get("sourceId");
-            String sourceProductLevel = getProductLevelFromSourceId(sourceId);
-            if (sourceProductLevel == null) {
-                throw new IllegalArgumentException("Invalid Source ID: " + sourceId);
+            if (!StringUtils.isEmpty(sourceId)) {
+                String sourceProductLevel = getProductLevelFromSourceId(sourceId);
+                if (sourceProductLevel == null) {
+                    throw new IllegalArgumentException("Invalid Source ID: " + sourceId);
+                }
+            }
+            String refsourceId = dsInfo.get("refSourceId");
+            if (!StringUtils.isEmpty(sourceId)) {
+                String sourceProductLevel = getProductLevelFromSourceId(refsourceId);
+                if (sourceProductLevel == null) {
+                    throw new IllegalArgumentException("Invalid Source ID: " + refsourceId);
```

Comment:
I think this is a copy/paste error.  
`if (!StringUtils.isEmpty(sourceId)) {`
should be
`if (!StringUtils.isEmpty(refsourceId)) {`